lib/Sema/TreeTransform.h

```diff
@@ -5815,6 +5815,26 @@ QualType TreeTransform<Derived>::TransformTypeVariableType(TypeLocBuilder &TLB,
   return TL.getType();
 }
 
+template<typename Derived>
+QualType TreeTransform<Derived>::TransformExistentialType(TypeLocBuilder &TLB,
+                                                          ExistentialTypeLoc TL) {
+  // TODO: be smarter about rebuilding these.
+  // TODO: the proper way to do the rebuilding is to have a RebuildExistentialType method. Add one?
+  auto *ExistType = TL.getTypePtr();
+  auto Quals = TL.getType().getQualifiers().getAsOpaqueValue();
+  auto TypeVar = QualType(ExistType->typeVar(), 0 /* Quals */);
+  QualType NewTypeVar = getDerived().TransformType(TypeVar);
+  if (NewTypeVar.isNull()) return QualType();
+  QualType NewInnerType = getDerived().TransformType(ExistType->innerType());
+  if (NewInnerType.isNull()) return QualType();
+  auto RawRes = SemaRef.ActOnExistentialType(SemaRef.Context, NewTypeVar.getTypePtr(), NewInnerType);
```

Comment:
You are missing the logic to avoid rebuilding here.   Most functions have the logic "getDerved(). AlwaysRebuild() || check that types are different".